packages/frontend/src/pages/applications.ts

```diff
@@ -17,13 +17,38 @@
  ***********************************************************************/
 
 import type { ApplicationState } from '@shared/src/models/IApplicationState';
-import type { Task } from '@shared/src/models/ITask';
 
-export interface ApplicationCell {
-  tasks?: Task[];
-  appState: ApplicationState;
-  recipeId: string;
-  modelId: string;
-  appPorts: number[];
-  modelPorts: number[];
+/* returns the status of the AI application, to be used by <IconStatus> */
+export function getApplicationStatus(
```

Comment:
should we have a new type being introduced for the return type

`'RUNNING' | 'DEGRADED' | 'STARTING' | 'USED' | 'DELETING' | ''`

also instead of '' should it be `UNKNOWN` (or another value)

IMHO it looks strange to have an empty state

Reply:
These values are the ones accepted by the component StatusIcon. I think we would need to define a type on the component first.

I'm using the empty string, as it is the default value for `status` of the StatusIcon component.

Reply:
I have made some changes in the latest commit.

I'm not sure if declaring this has a sense? (we want to accept all values, but some of them are "privileged", as they will have specific effect on the component)
```
export let status: knownStatus | string = '';
```

Reply:
well "anything that is a string" | string is just a string

it's like `"foo" | string | any` at the end it's any that is winning

so IMHO either you put 'KnownStatus" or just string 

using KnownStatus | string people will never know that there are like "privilege" components

or you have a component like `<KnownStatusComponent>` accepting only `KnownStatus` but that will use `<StatusComponent>` "behind the scene"